.github/workflows/spack_cpu_build.yaml

```diff
@@ -0,0 +1,52 @@
+name: Spack Builds
+
+# Until we remove the need to clone submodules to build, this should on be in PRs
+on: [pull_request]
+
+jobs:
+  exago_spack_builds:
+    # 20.04 is a version shared by E4S cache and Spack binaries for x86_64
+    runs-on: ubuntu-20.04
+    container: spack/ubuntu-focal:latest
+    strategy:
+      matrix:
+        spack_spec:
+          - exago@develop+mpi~ipopt~hiop~logging~python~raja ^openmpi
+          - exago@develop~mpi~ipopt+hiop~logging~python~raja
+          - exago@develop~mpi~ipopt~hiop~logging~python~raja
+          - exago@develop~mpi~ipopt~hiop~logging+python~raja
+          - exago@develop~mpi~ipopt+hiop~logging~python+raja
+          - exago@develop~mpi~ipopt~hiop~logging~python+raja
```

Comment:
@abhyshr we are going to build out this matrix through things like Spack CI and connecting GitLab runners to AWS, so this is just a first pass, but what do you think of the usefulness of this set of builds? Are there any that you would cut?

cc @Jayapreethi and @paulrigor

Reply:
For all our scopflow and sopflow runs, IPOPT is a mandatory requirement. So, I think you can condense this to 
+ipopt+python
+ipopt+hiop+python
+ipopt+hiop+raja+python

is there a reason you want also want to do tests without mpi? All our applications use MPI so why the need for no MPI tests?

We also need testing with different optimization options (debug, release)

Reply:
These tests aren't actually running anything at the moment. In the future, we might do a small `spack test exago`, but for now this is just to catch build failures in configurations currently not built in CI.

We can build everything with MPI, but that would drastically increase the build size, and so I don't want more than one MPI build here for the moment without good reason.

We can't do `+ipopt` in CI here since we would need the CoinHSL tarball in GitHub actions/our repo, which is against the license. So we can't have any of the builds here be `+ipopt`.

Reply:
Building with a few different release types is a good suggestion. I'll try and add that on Monday